txhelpers/txhelpers.go

```diff
@@ -241,30 +244,55 @@ func TxInvolvesAddress(msgTx *wire.MsgTx, addr string, c VerboseTransactionGette
 // indexes and the corresponding previous outpoints determined.
 func TxConsumesOutpointWithAddress(msgTx *wire.MsgTx, addr string,
 	c VerboseTransactionGetter, params *chaincfg.Params) (prevOuts []PrevOut, prevTxs []*TxWithBlockData) {
+	// Send all the raw transaction requests
+	type promiseGetRawTransaction struct {
+		result rpcclient.FutureGetRawTransactionVerboseResult
+		prevtx chainhash.Hash
+	}
+	numPrevOut := len(msgTx.TxIn)
+	promisesGetRawTransaction := make([]promiseGetRawTransaction, numPrevOut)
+
+	for inIdx, txIn := range msgTx.TxIn {
+		hash := &txIn.PreviousOutPoint.Hash
+		if zeroHash.IsEqual(hash) {
+			continue // coinbase or stakebase
+		}
+		promisesGetRawTransaction[inIdx] = promiseGetRawTransaction{
+			result: c.GetRawTransactionVerboseAsync(hash),
+			prevtx: *hash,
+		}
+	}
+
 	// For each TxIn of this transaction, inspect the previous outpoint.
 	for inIdx, txIn := range msgTx.TxIn {
```

Comment:
range over promisesGetRawTransaction and I you can ditch the zeroHash.IsEqual check below since it is done above.

Reply:
I suppose we can do that, but we're checking `prevOut := &txIn.PreviousOutPoint` so we'll need to either store those in the first loop is eliminate the checks in the second.

Reply:
Good suggestion.  I had another shot and came up with https://github.com/decred/dcrdata/pull/1144/commits/f5de2c6109544281ab946b62f31a2bcbb4515302